layer/components/NumInput.vue

```diff
@@ -1,121 +1,77 @@
-<script lang="ts" setup>
-import { useDebounceFn } from '@vueuse/core'
-import { BigNumberInBase } from '@injectivelabs/utils'
-import {
-  stripNonDigits,
-  validateNumericInput,
-  convertToNumericValue
-} from './../utils/input'
-import { KeydownEvent, PasteEvent } from './../types'
+<script setup lang="ts">
+import { useIMask } from 'vue-imask'
+import type { FactoryOpts } from 'imask'
 
 const props = defineProps({
-  isShowMask: Boolean,
+  autofix: Boolean,
+  thousandsSeparator: Boolean,
 
   modelValue: {
-    type: [String, Number],
+    type: String,
     default: ''
   },
 
-  maxDecimals: {
+  decimals: {
     type: Number,
-    default: 6
+    default: 18
+  },
+
+  max: {
+    type: Number,
+    // eslint-disable-next-line
+    default: 9999999999999999999
```

Comment:
We should not set the min/max as required, and this hardcoded value is should not be here, what if they enter `99999999999999999999`?

Reply:
for some reason, the default max is set to `10**13` that is why explicitly have to set it here, i can set it to i higher value but i think 18 significant places is enough, and also we have 18 (or more) decimal places. 
and both are not required.
default for max is 9999.
default for min is undefined

Reply:
let's set max as 10 ** 18

also this component should be ui-shared folder not layer, else we wont be able to use it on hub and helix until we refactor them to use `layer`